public/static/js/distros.js

```diff
@@ -195,24 +158,73 @@ mciModule.controller('DistrosCtrl', function($scope, $window, $location, $anchor
     var distroHash = $location.hash();
     if (distroHash) {
       // If the distro exists, load it.
-      var distro = $scope.getDistroById(distroHash);
-      if (distro) {
-    $scope.activeDistro = distro;
-    return;
+      if (distroHash == newId && $scope.tempDistro) {
+          $scope.activeDistro = $scope.tempDistro;
+      } else if (distroHash != newId) {
+        $scope.getDistroById(distroHash).then(function(distro) {
+          if (distro) {
+            $scope.readOnly = (!$window.aclEnabled && !$window.isSuperUser) || ($window.aclEnabled && distro.permissions.distro_settings < 20)
+            $scope.activeDistro = distro.distro;
+          }
+        });
       }
+    } else {
+      // default to first id
+      $scope.setActiveDistroId($scope.distroIds[0]);
```

Comment:
This still has the danger of going into an infinite loop if there's a problem loading the first ID. I think to prevent that, we just do nothing if this is the first one

Reply:
There won't be an infinite loop, if there are no ids then this will never get hit. If the first ID no longer exists for some reason, then this function will get called again but this branch of the if/else statement won't get hit. This is required for when the page is first loaded.